lib/ansible/module_utils/vmware.py

```diff
@@ -213,6 +213,16 @@ def find_host_portgroup_by_name(host, portgroup_name):
     return None
 
 
+def _get_vm_prop(vm, expression):
+    """Safely get a property or return None"""
+    result = None
+    try:
+        result = eval(expression)
+    except:
+        pass
+    return result
```

Comment:
@Akasurde I've tested this patch on cloning from snapshots, so I think it does work to some degree. Can you extrapolate on which situations you don't think it will work?